### ISSUE TITLE
[safer-cluster] `node_pools` Does Not Accept Complex Settings

### ISSUE THREAD

**Issue by goakley** [terraform-google-modules/terraform-google-kubernetes-engine#1711]
### TL;DR

[`safer-cluster`](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/safer-cluster/variables.tf)'s `node_pools` variable is of type `list(map(string))`, while the underlying [`beta-private-cluster`](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/beta-private-cluster/variables.tf) that powers `safer-cluster` accepts `list(map(any))`.  This prevents us from specifying node pool settings [such as autoscaling](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/beta-private-cluster/cluster.tf#L482C1-L491), since doing so requires passing nested maps to the `node_pools` variable.

### Expected behavior

It should be possible to specify `safer-cluster` node pools with additional complex properties, such as autoscaling:
```
node_pools = [{
  name              = "default-node-pool"
  machine_type      = var.machine_type
  node_locations    = join(",", var.zones)
  max_pods_per_node = 32
  image_type        = "COS_CONTAINERD"
  min_count         = 4
  max_count         = 32
  autoscaling = {
    location_policy = "BALANCED"
  }
}]
```

### Observed behavior

Attempting to apply Terraform with `safer-cluster` `node_pools` with autoscaling results in an error:

```
Error: Invalid value for module argument
...
The given value is not suitable for child module variable "node_pools" defined at `<...>/modules/gke.gke/modules/safer-cluster/variables.tf:144,1-22: element 1: element "autoscaling": string required.
```

This is because `safer-clusters` requires the map values to ONLY be strings (which is not what the underlying `beta-private-cluster` module expects).

### Terraform Configuration

```hcl
module "gke" {
  source  = "terraform-google-modules/kubernetes-engine/google//modules/safer-cluster/"
  version = "24.1.0"

  project_id                 = var.project_id
  name                       = "gke-1"
  regional                   = true
  region                     = "us-east1"
  zones                      = ["us-east1-b"]
  network                    = var.network_name
  subnetwork                 = var.network_name
  ip_range_pods              = "gke-1-pods"
  ip_range_services          = "gke-1-svc"
  master_ipv4_cidr_block     = "172.16.0.0/30"
  enable_private_endpoint    = false
  add_cluster_firewall_rules = true
  datapath_provider = "DATAPATH_PROVIDER_UNSPECIFIED"
  firewall_inbound_ports = ["8443", "9443"]
  maintenance_start_time = "00:00"

  authenticator_security_group = var.authenticator_security_group

  grant_registry_access = false

  master_authorized_networks = concat([
    {
      display_name = var.network_name
      cidr_block   = var.master_authorized_networks_cidr_block
    }
  ])

  cloudrun = false

  node_pools = [
    {
      name              = "default-node-pool"
      machine_type      = "c2-standard-8"
      node_locations    = "us-east1-b"
      max_pods_per_node = 32
      image_type        = "COS_CONTAINERD"
      min_count         = 4
      max_count         = 32
      autoscaling = {
        location_policy = "BALANCED"
      }
    }
  ]
  node_pools_labels = {
    all = {}
  }
  node_pools_metadata = {
    all = {}
  }
  node_pools_taints = {
    all = []
  }
  node_pools_tags = {
    all = []
  }

  cluster_autoscaling = {
    enabled             = false
    autoscaling_profile = "OPTIMIZE_UTILIZATION"
    max_cpu_cores       = 0
    min_cpu_cores       = 0
    max_memory_gb       = 0
    min_memory_gb       = 0
    gpu_resources       = []
  }
}
```
```


### Terraform Version

```sh
Terraform v0.15.5
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.44.1
+ provider registry.terraform.io/hashicorp/google-beta v4.44.1
```
```


### Additional information

_No response_

**Comment by alyssahardy** [comment]
I encountered issues as well on beta-private-cluster, which uses list(map(any)). I think autoscaling needs to be treated like metadata or like the guest_accelerator config. Changing it from list(map(string)) to list(map(any)) won't do any good, as the any just makes Terraform find the type from the content. [TF docs on any type constraint with collections](https://developer.hashicorp.com/terraform/language/expressions/type-constraints#any-with-collection-types)

**Comment by alyssahardy** [comment]
This is still an issue
